tests/core/full_node/stores/test_coin_store.py

```diff
@@ -239,19 +239,44 @@ async def test_rollback(self, cache_size: uint32, db_version, bt):
                             removals,
                         )
 
-                    coins = block.get_included_reward_coins()
+                    coins = list(block.get_included_reward_coins())
                     records = [await coin_store.get_coin_record(coin.name()) for coin in coins]
 
-                    await coin_store._set_spent([r.name for r in records], block.height)
+                    spend_selected_coin = selected_coin is not None
+                    if block.height != 0 and selected_coin is None:
+                        # Select one coin which will be spent later
+                        selected_coin = records[0]
+                        await coin_store._set_spent([r.name for r in records[1:]], block.height)
+                    else:
+                        await coin_store._set_spent([r.name for r in records], block.height)
+
+                    if spend_selected_coin:
+                        assert selected_coin is not None
+                        await coin_store._set_spent([selected_coin.name], block.height)
 
                     records = [await coin_store.get_coin_record(coin.name()) for coin in coins]
                     for record in records:
                         assert record is not None
-                        assert record.spent
-                        assert record.spent_block_index == block.height
+                        if (
+                            selected_coin is not None
+                            and selected_coin.name == record.name
+                            and not selected_coin.confirmed_block_index < block.height
+                        ):
+                            assert not record.spent
+                        else:
+                            assert record.spent
+                            assert record.spent_block_index == block.height
 
-            reorg_index = 8
-            await coin_store.rollback_to_block(reorg_index)
+                    if spend_selected_coin:
+                        break
+
+            assert selected_coin is not None
+            reorg_index = selected_coin.confirmed_block_index
+
+            # The reorg will revert the creation and spend of many coins. It will also revert the spend (but not the
+            # creation) of the selected coin.
+            changed_records = await coin_store.rollback_to_block(reorg_index)
+            assert selected_coin in changed_records
```

Comment:
maybe I'm misunderstanding this test; but it seems to me we would need to ensure `changed_records` contains exactly the coins we expect to have been reverted by the rollback

Reply:
We could do that but it would make the test a little more fragile, ie if we changed the blocks then the result will be different, also it's more work.

However it's probably a good idea to do that to make the test more useful

Reply:
Just added more checks